File: sportsbook-app/betting-service/src/main/java/com/sportsbook/bettingservice/domain/Bet.java
package com.sportsbook.bettingservice.domain;

import jakarta.persistence.*;
import lombok.*;
import lombok.experimental.Accessors;

import java.time.Instant;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Accessors(chain = true)
@Entity
@Table(name = "bets")
public class Bet {
    @Id
    private String id;
    private String playerId;
    private String userId;
    private String eventId;
    private String marketId;
    private String outcomeId;
    private double stake;
    private double odds;
    private double stakeCents;
    @Enumerated(EnumType.STRING)
    private BetStatus status;
    private Instant createdAt;
    private Instant updatedAt;
}

File: sportsbook-app/betting-service/src/main/kotlin/com/sportsbook/bettingservice/domain/Bet.kt
package com.sportsbook.bettingservice.domain

import jakarta.persistence.Entity
import jakarta.persistence.EnumType
import jakarta.persistence.Enumerated
import jakarta.persistence.Id
import jakarta.persistence.Table
import java.time.Instant

@Entity
@Table(name = "bets")
class Bet {
    @Id
    var id: String? = null
    var playerId: String? = null
    var userId: String? = null
    var eventId: String? = null
    var marketId: String? = null
    var outcomeId: String? = null
    var stake: Double = 0.0
    var odds: Double = 0.0
    var stakeCents: Double = 0.0

    @Enumerated(EnumType.STRING)
    var status: BetStatus? = null
    var createdAt: Instant? = null
    var updatedAt: Instant? = null
}
